python/src/linalg.cpp

```diff
@@ -0,0 +1,218 @@
+
+// Copyright © 2023 Apple Inc.
+
+#include <limits>
+#include <numeric>
+#include <ostream>
+#include <string>
+#include <variant>
+
+#include <pybind11/iostream.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include "mlx/linalg.h"
+#include "mlx/ops.h"
+#include "mlx/utils.h"
+
+#include "python/src/load.h"
+#include "python/src/overloaded.h"
+#include "python/src/utils.h"
+
+namespace py = pybind11;
+using namespace py::literals;
+
+using namespace mlx::core;
+using namespace mlx::core::linalg;
+
+void init_linalg(py::module_& parent_module) {
+  auto m =
+      parent_module.def_submodule("linalg", "mlx.core.linalg: Linear Algebra.");
+
+  m.def(
+      "norm",
+      [](const array& a,
+         const std::variant<std::monostate, int, double, std::string>& ord,
+         const std::variant<std::monostate, int, std::vector<int>>& axis,
+         const bool keepdims,
+         const StreamOrDevice stream) {
+        return std::visit(
+            overloaded{
+                [&](const double p) {
+                  if (std::isinf((float)p) || std::isinf(p)) {
```

Comment:
Is the cast to float here with the first `isinf` necessary?

Reply:
I am not sure. Ideally, we should handle infinity for every 'float' data type, regardless of the precision. I think that Python bindings map infinity to float32 infinity.